src/common/TrackedOp.cc

```diff
@@ -22,16 +22,69 @@ static ostream& _prefix(std::ostream* _dout)
   return *_dout << "-- op tracker -- ";
 }
 
+void OpHistoryServiceThread::BreakThread() {
+  queue_spinlock.lock();
+  _external_queue.clear();
+  _break_thread = true;
+  queue_spinlock.unlock();
+}
+
+void OpHistoryServiceThread::InsertOp(utime_t& now, TrackedOpRef op) {
+  queue_spinlock.lock();
+  _external_queue.emplace_back(now, op);
```

Comment:
This allocates, right? Is using a mutex really not okay here?

Reply:
I wanted threads to wait for lock to be freed without being preempted. See the graph, there's a slight difference in how it affects machinery performance.